pyqtgraph/parametertree/interactive.py

```diff
@@ -0,0 +1,463 @@
+        extra parameters can be defined here if the original function uses ``**`` to consume additional keyword
+        arguments. Each override can be a value (e.g. 5) or a dict specification of a parameter
+        (e.g. dict(type='list', limits=[0, 10, 20]))
+    """
+    funcDict = funcToParamDict(func, title=title, **overrides)
+    children = funcDict.pop('children', []) # type: list[dict]
+    chNames = [ch['name'] for ch in children]
+    parent = _resolveParent(parent, nest, funcDict, existOk)
+
+    toExec = runFunc or func
+    if not isinstance(toExec, InteractiveFunction):
+        toExec = InteractiveFunction(toExec)
+
+    # Values can't come both from deferred and overrides/params, so ensure they don't get created
+    if ignores is None:
+        ignores = []
+    ignores = list(ignores) + list(toExec.deferred)
+
+    # Recycle ignored content that is needed as a value
+    recycleNames = set(overrides) & set(ignores) & set(chNames)
+    extra = {key: overrides[key] for key in recycleNames}
+    toExec.extra.update(extra)
+
+    useParams = []
+    for chDict in children:
+        name = chDict['name']
+        if name in ignores:
+            continue
+        child = _createFuncParamChild(parent, chDict, runOpts, existOk, toExec)
+        useParams.append(child)
+
+    toExec.setParams(useParams)
+    ret = parent
+    if RunOpts.ON_BUTTON in runOpts:
+        # Add an extra button child which can activate the function
+        button = _makeRunButton(nest, funcDict.get('tip'), toExec)
+        # Return just the button if no other params were allowed
+        if not parent.hasChildren():
+            ret = button
+        parent.addChild(button, existOk=existOk)
+
+    # Keep reference to avoid `toExec` getting garbage collected and allow later access
+    return ret
+
+def _resolveParent(parent, nest, parentOpts, existOk):
+    if parent is None or nest:
+        host = Parameter.create(**parentOpts)
+        if parent is not None:
+            # Parent was provided and nesting is enabled, so place created args inside the nested GroupParmeter
+            host = parent.addChild(host, existOk=existOk)
+        parent = host
+    return parent
+
+def _createFuncParamChild(parent, chDict, runOpts, existOk, toExec):
+    name = chDict['name']
+    # Make sure args without defaults have overrides
+    if chDict['value'] is RunOpts.PARAM_UNSET and name not in toExec.deferred and name not in toExec.extra:
+        raise ValueError(f'Cannot interact with "{toExec} since it has required parameter "{name}"'
+                         f' with no default or deferred value provided.')
+    child = parent.addChild(chDict, existOk=existOk)
+    # I tried connecting directly to the runnables in `toExec`, but they result in early garbage collection. This
+    # doesn't happen with local functions
+    def runner(param, val):
+        toExec.runFromChangedOrChanging(param, val)
```

Comment:
@j9ac9k This prevents garbage collection of interactive functions created at line 393. Is there a better way of making sure the `runFrom...` function of `toExec` is called without an eternal dangling reference here?

Reply:
for something like that preventing garbage collection, the only other thing I can think of is having some kind of dictionary being kept in the global namespace that holds onto them.  I think this is a fair use of `global` which I know most people shy away from, but for something like this, I think it's fair.

Reply:
Would it be fair to assume InteractiveFunction should share the same life cycle as the initial function? If so, I could do this instead:

```python
func.interactiveRef = toExec
```

I think this would only fail for lambdas defined in the call to `interact`. In any case, I like it a lot better than an inaccessible reference inside a function call...

Reply:
from what I've gathered form this, I would assume this is a fair assumption.